org.sonarlint.eclipse.cdt/src/org/sonarlint/eclipse/cdt/internal/BuildWrapperJsonFactory.java

```diff
@@ -0,0 +1,139 @@
+package org.sonarlint.eclipse.cdt.internal;
+
+import java.util.Map;
+import org.eclipse.cdt.core.parser.IScannerInfo;
+
+public class BuildWrapperJsonFactory {
+  private static final String COMPILER = "clang";
+  private static final String EXE_ID = "uniqueid";
+
+  public String create(Map<String, IScannerInfo> fileInfo, String baseDirPath) {
+    StringBuilder builder = new StringBuilder();
+    builder.append("{"
+      + "\"version\":0,"
+      + "\"captures\":[");
+
+    writeCompiler(builder);
+    builder.append(",");
+    writeCompiler(builder);
+
+    for (Map.Entry<String, IScannerInfo> f : fileInfo.entrySet()) {
+      builder.append(",");
+      writeFile(builder, baseDirPath, f.getKey(), f.getValue().getIncludePaths(), f.getValue().getDefinedSymbols());
+    }
+
+    builder.append("]}");
+    return builder.toString();
+  }
+
+  private static void writeFile(StringBuilder builder, String baseDirPath, String relativeFilePath, String[] includes,
+    Map<String, String> symbols) {
+    builder.append("{"
+      + "\"compiler\":\"" + COMPILER + "\","
+      + "\"cwd\":" + quote(baseDirPath) + ","
+      + "\"executable\":\"" + EXE_ID + "\","
+      + "\"cmd\":["
+      + "\"clang\"");
+
+    writeSymbols(builder, symbols);
+    writeIncludes(builder, includes);
+    builder
+      .append(",\"" + relativeFilePath + "\"")
+      .append("]}");
+
+  }
+
+  private static void writeIncludes(StringBuilder builder, String[] includes) {
+    for (String include : includes) {
+      builder.append(",\"-I\"," + quote(include) + "");
+    }
+  }
+
+  private static void writeSymbols(StringBuilder builder, Map<String, String> symbols) {
+    for (Map.Entry<String, String> symbol : symbols.entrySet()) {
+      builder.append(",");
+      builder.append(quote("-D" + symbol.getKey() + "=" + symbol.getValue()));
+    }
+  }
+
+  private static void writeCompiler(StringBuilder builder) {
+    builder.append("{"
+      + "\"compiler\":\"" + COMPILER + "\","
+      + "\"executable\":\"" + EXE_ID + "\","
+      + "\"stdout\":\"\","
+      + "\"stderr\":\"\""
+      + "}");
+  }
+
+  /**
+   * Copied from Jettison
+   */
+  public static String quote(String string) {
+    if (string == null || string.length() == 0) {
+      return "\"\"";
+    }
+
+    char c = 0;
+    int i;
+    int len = string.length();
+    StringBuilder sb = new StringBuilder(len + 4);
+    String t;
+
+    sb.append('"');
+    for (i = 0; i < len; i += 1) {
+      c = string.charAt(i);
+      switch (c) {
+        case '\\':
+        case '"':
+        case '/':
+          sb.append('\\');
+          sb.append(c);
+          break;
+        case '\b':
+          sb.append("\\b");
+          break;
+        case '\t':
+          sb.append("\\t");
+          break;
+        case '\n':
+          sb.append("\\n");
+          break;
+        case '\f':
+          sb.append("\\f");
+          break;
+        case '\r':
+          sb.append("\\r");
+          break;
+        default:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)